fix autograd bug re: inclusion extraneous field mnts

tylerflex opened the pull request:
still not sure:
this isn't triggered by regular autograd code. might be something with the metrics

tylerflex committed:
fix autograd inclusion extraneous field mnts

yaugenst-flex commented on tidy3d/components/data/monitor_data.py:
```diff
@@ -1093,6 +1093,10 @@ def to_adjoint_field_sources(self, fwidth: float) -> List[CustomCurrentSource]:
                     src_field_components[name] = ScalarFieldDataArray(values, coords=coords)
 
+            # dont include this source if no data
+            if all(fld_cmp is None for _, fld_cmp in src_field_components.items()):
```
`src_field_components.values()`?

tylerflex replied:
I think pylint used to complain about iterating over values, so I got in the habit of always doing over `items()`. but not really a strong preference.

yaugenst-flex replied:
hm I don't see a reason for this. might have been an artifact of possibly unordered dicts, but that's not the case anymore as of python 3.7..

yaugenst-flex replied:
e.g. ruff will complain by default about iterating over items if only keys or values are used. we might have disabled that rule 🤔

tylerflex commented:
fixed values() iteration but still am not able to reproduce this error. (basically the test I added passes even without the change) would be good (but not super high priority) to understand why exactly this popped up and reproduce it in a test

yaugenst-flex commented:
yeah I also can't reproduce it. I think it's because emulated run does not exactly reproduce what is happening in regular run, because the code will fail is using regular run

momchil-flex commented:
Merging but this would need further investigation
